wandb/cli/cli.py

```diff
@@ -1519,6 +1519,351 @@ def scheduler(
+
+    for job in jobs:
+        for version in job["edges"]:
+            name = version["node"]["artifactSequence"]["name"]
+            full_name = f"{entity}/{project}/{name}"
+            aliases = [x["alias"] for x in version["node"]["aliases"]]
+            if len(aliases) == 1:
+                # no custom alias, just verison, skip
+                continue
+            print(full_name, aliases)
+
+
+def _make_git_data(path):
+    """mock, to be replaced"""
+    remote = path
+    commit = "bdb06b4b5b6449d891ca7d019f7885394d57fcfb"
+    requirements = ["wandb"]
+    python = "3.11.4"
+    return remote, commit, requirements, python
+
+
+def _make_code_artifact_name(path: str, name: Optional[str]) -> str:
+    """Make a code artifact name from a path and user provided name"""
+    if name:
+        return f"code-{name}"
+
+    if len(path) > 7:
+        if path[-1] == "/":
+            path = path[:-1]
+        path_name = f"code-{path.replace('/', '-')}"
+        wandb.termlog(f"No name provided, using path for name: {path_name}")
+        return path_name
+
+    generated_name = f"code-{wandb.util.generate_id()}"
+    wandb.termlog(f"No name provided, generating name: {generated_name}")
+    return generated_name
+
+
+@job.command()
+@click.argument("job")
+def describe(job):
+    wandb.termlog(f"Describing job: {job}")
+
+    public_api = PublicApi()
+    job = public_api.job(name=job)
+
+    wandb.termlog(f"{job._job_info}")
+
+
+@job.command(
+    no_args_is_help=True,
+)
+@click.option(
+    "--project",
+    "-p",
+    envvar=env.PROJECT,
+    help="The project you want to list jobs from.",
+)
+@click.option(
+    "--entity",
+    "-e",
+    default="models",
+    envvar=env.ENTITY,
+    help="The entity the jobs belong to",
+)
+@click.option(
+    "--name",
+    "-n",
+    help="Name for the job",
+)
+@click.option(
+    "--description",
+    "-d",
+    help="Description for the job",
+)
+@click.option(
+    "--type",
+    "-t",
+    "_type",
+    type=click.Choice(("repo", "artifact", "image")),
+    default="artifact",
+    help="Type of job to create, defaults to artifact",
+)
+@click.option(
+    "--alias",
+    "-a",
+    "aliases",
+    help="Alias for the job",
+    multiple=True,
+)
+@click.option(
+    "--entry-point",
+    "-en",
+    "entrypoint",
+    help="Codepath to the main script, required for repo jobs",
+)
+@click.argument("path")
+def create(path, project, entity, name, _type, description, aliases, entrypoint):
+    """Create a job from a source, without a wandb run.
+
+    Jobs can be of three types, repo, artifact, or image.
+
+    repo: A git source, with an entrypoint pointing to the main python executable.
+    artifact: A code path, containing a requirements.txt file.
+    image: A docker image.
+    """
+    api = _get_cling_api()
+    entity = entity or os.getenv("WANDB_ENTITY")
+    if not entity:
+        wandb.termerror("No entity provided")
+        return
+
+    project or os.getenv("WANDB_PROJECT")
```

Comment:
Should this be `project = project or os.getenv(...`?

Reply:
🤦🏼‍♂️